tracing-jersey/src/main/java/com/palantir/tracing/jersey/TraceEnrichingFilter.java

```diff
@@ -102,12 +103,12 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
     }
 
     // Returns true iff the context contains a "1" X-B3-Sampled header, or absent if there is no such header.
-    private static Optional<Boolean> hasSampledHeader(ContainerRequestContext context) {
+    private static Observability getObservabilityFromHeader(ContainerRequestContext context) {
         String header = context.getHeaderString(TraceHttpHeaders.IS_SAMPLED);
         if (header == null) {
-            return Optional.empty();
+            return Observability.UNDECIDED;
         } else {
-            return Optional.of(header.equals("1"));
+            return header.equals("1") ? Observability.SAMPLE : Observability.DO_NOT_SAMPLE;
```

Comment:
pedantic nit: would prefer `"1".equals(header)`, though it won't make a difference in practice here since we've already validated non-null